src/components/ItemList.svelte

```diff
@@ -38,7 +49,10 @@
   // (above)
   $: {
     // filter items by search terms and expiry state
-    filteredItems = filterItems(items, search, showExpired);
+    filteredItems =
+      (filterItems(items, search, showExpired).length &&
+        filterItems(items, search, showExpired)) ||
+      fullTextSearch(search);
```

Comment:
So the logic here is that if searching by name/type doesn't return anything, we then try searching by description. Should I combine the 2 results instead of doing this way?

Reply:
I think we should use what's called a "document" index in flexsearch, and search by name/type/description all at once (then filter by `showExpired` at the very end). I played around with this a bit, here's an example of a document index:

https://irydium.dev/repl?id=b9c36dec-0418-11ec-866c-0a4e9a1089db